src/routes/settings/ManageFederations.tsx

```diff
@@ -0,0 +1,219 @@
+                    name="federation_code"
+                    validate={[
+                        required(
+                            i18n.t(
+                                "settings.manage_federations.federation_code_required"
+                            )
+                        )
+                    ]}
+                >
+                    {(field, props) => (
+                        <TextField
+                            {...props}
+                            {...field}
+                            error={field.error}
+                            label={i18n.t(
+                                "settings.manage_federations.federation_code_label"
+                            )}
+                            placeholder="fedi1..."
+                            required
+                        />
+                    )}
+                </Field>
+                <Button
+                    loading={feedbackForm.submitting}
+                    disabled={!feedbackForm.touched || feedbackForm.invalid}
+                    intent="blue"
+                    type="submit"
+                >
+                    {i18n.t("settings.manage_federations.add")}
+                </Button>
+                <Show when={error()}>
+                    <InfoBox accent="red">{error()?.message}</InfoBox>
+                </Show>
+                <Show when={success()}>
+                    <InfoBox accent="green">{success()}</InfoBox>
+                </Show>
+            </VStack>
+        </Form>
+    );
+}
+
+function FederationListItem(props: { fed: MutinyFederationIdentity }) {
+    const i18n = useI18n();
+    const [state, actions] = useMegaStore();
+
+    async function removeFederation() {
+        setConfirmLoading(true);
+        try {
+            await state.mutiny_wallet?.remove_federation(
+                props.fed.federation_id
+            );
+            await actions.refreshFederations();
+        } catch (e) {
+            console.error(e);
+        }
+        setConfirmLoading(false);
+    }
+
+    async function confirmRemove() {
+        setConfirmOpen(true);
+    }
+
+    const [confirmOpen, setConfirmOpen] = createSignal(false);
+    const [confirmLoading, setConfirmLoading] = createSignal(false);
+
+    return (
+        <>
+            <FancyCard>
+                <Show when={props.fed.federation_name}>
+                    <header class={`font-semibold`}>
+                        {props.fed.federation_name}
+                    </header>
+                </Show>
+                <Show when={props.fed.welcome_message}>
+                    <p>{props.fed.welcome_message}</p>
+                </Show>
+                <Show when={props.fed.federation_expiry_timestamp}>
+                    <KeyValue
+                        key={i18n.t("settings.manage_federations.expires")}
+                    >
+                        <time>
+                            {timeAgo(props.fed.federation_expiry_timestamp)}
```

Comment:
`timeAgo` isn't the best name for this function now since it is returning a future time, but fine for now i guess

Reply:
Would be good to display a warning or something if the expiration is in the past though